cstore_writer.c

```diff
@@ -191,12 +173,43 @@ CStoreBeginWrite(const char *filename, CompressionType compressionType,
 	writeState->stripeWriteContext = stripeWriteContext;
 	writeState->blockDataArray = blockData;
 	writeState->compressionBuffer = NULL;
+	writeState->activeBlockNumber = blockNumber;
+	writeState->relation = relation;
 
 	return writeState;
 }
 
 
 /*
+ * EnsureDataForkExists checks if data fork exists for writing data, and
+ * creates if it is not present. Relation must be previously opened with
+ * exclusive lock (AccessExclusive or ShareUpdateExclusive).
+ */
+static void
+EnsureDataForkExists(Relation relation, bool logging)
+{
+	SMgrRelation srel = NULL;
+
+	RelationOpenSmgr(relation);
+	srel = (SMgrRelation) relation->rd_smgr;
+
+	if (!smgrexists(srel, DATA_FORKNUM))
+	{
+		smgrcreate(srel, DATA_FORKNUM, false);
+
+		if (logging)
+		{
+			log_smgrcreate(&srel->smgr_rnode.node, DATA_FORKNUM);
+		}
+	}
+
+	Assert(smgrexists(srel, DATA_FORKNUM));
+
+	RelationCloseSmgr(relation);
```

Comment:
we won't be needing it until flush stripe.  

are we supposed to keep it open ?